### PR TITLE
updating the project api

### PR DESCRIPTION
**Pull request by a1astair**
adding in default legislation logic to publish, unpublish and get project call

### CODE REVIEW

File: api/helpers/actions.js [hunk 1]
```diff
@@ -23,14 +30,21 @@ exports.isPublished = async function (o) {
...
         // Object wasn't already published?
-        if (o.read.includes('public')) {
-            var newReadArray = o.read.filter(perms => perms !== 'public');
-            o.read = newReadArray;
-            // Remove publish, save then return.
-            resolve(o.save());
+        let newReadArray;
+        if (isProject && o[o.currentLegislationYear].read.includes('public')) {
+          newReadArray = o[o.currentLegislationYear].read;
+          newReadArray.push('public');
```
Comment (MaxWardle) [inline api/helpers/actions.js]: Shouldn't this remove 'public'

File: api/helpers/actions.js [hunk 2]
```diff
@@ -2,17 +2,24 @@
...
 
-exports.publish = async function (o) {
+exports.publish = async function (o, isProject = false) {
     return new Promise(function (resolve, reject) {
+      // Need project specific logic to handle legislation keys
         // Object wasn't already published?
-        if (!o.read.includes('public')) {
-            var newReadArray = o.read;
+        let newReadArray;
+        if (isProject && !o[o.currentLegislationYear].read.includes('public')) {
```
Comment (MaxWardle) [inline api/helpers/actions.js]: I think the currentLegislationYear in the root is a string of project data keys, i.e. "legislation_2018", etc.. isn't it? I think we'll have to  pull out the year from the string here.

Comment (a1astair) [reply api/helpers/actions.js]: we are using the currentLegislationYear as the key here to access the read array inside of o.legislation_2018 or whichever legislation is current

File: api/helpers/actions.js (hunk 1 above)
Comment (a1astair) [reply api/helpers/actions.js]: whoops good catch!

File: api/helpers/actions.js (hunk 2 above)
Comment (MaxWardle) [reply api/helpers/actions.js]: right, my bad

**Comment by marklise** [comment]
Publish, Unpublish should be generic and not have any specific logic around legislation in it.  This should be done a different way.

**Pull request merged**